var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -69,6 +69,12 @@ def install(self, spec, prefix):
             '--with-blas-libs=%s' % ' '.join(blas.names),
             '--with-blas-lib-dirs=%s' % ' '.join(blas.directories)
         ]
+        
+        if spec.satisfies('^netlib-lapack'):
+            # netlib-lapack puts the libs in lib64
+            configure_args.extend([
+                "--with-lapack-lib-dirs=%s/lib64".join(lapack.directories),
```

Comment:
that's exactly what should happen above. It's the wrong place to fix. If you `netlib-lapack` lives in `lib64`, you need to fix that in `netlib-lapack` package.